test_pull_requests

```diff
@@ -456,32 +463,43 @@ class InvalidatingGitHubCache < Faraday::HttpCache
     def write(request, response)
       super
       key = cache_key_for(request.url)
-      url = synthesize_url(request.url)
-      @invalidatable[url.to_s][key] = true
+      # url = derive_api_match_url(request.url)
+      @invalidatable[request.url.to_s].add key
+    end
+
+    # Deserialize cached (request, response) entries for the given key
+    # and yield them to the given block for modification, then
+    # re-serialize them and add them back to the cache
+    def modify_cache_entries_for_key(cache_key)
+      entries = cache.read(cache_key)
+      entries = entries.dup if entries.frozen?
+      if entries
+        entries = entries.map { |entry| deserialize_entry(*entry) }
+        entries.each do |entry_req, entry_res|
+          yield entry_req, entry_res
+          write(Faraday::HttpCache::Request.new(entry_req), Faraday::HttpCache::Response.new(entry_res))
+        end
+      end
     end
 
-    # We invalidate more than is strictly necessary to handle e.g.:
+    # We may invalidate more than is strictly necessary, to handle
+    # instances where it's difficult to efficiently link a destructive
+    # request to the specific impacted cached entries, e.g.:
+    #
     #  DELETE https://api.github.com/repos/user/repo/issues/comments/111111111
     #    should invalidate: https://api.github.com/repos/user/repo/issues/1/comments?per_page=100
-    def invalidate_urls(url=nil)
-      iurls=[]
-      if url
-        base_url = synthesize_url(url)
-        iurls = @invalidatable.keys.select { |iurl| iurl.to_s.start_with? base_url.to_s }
+    def invalidate_cache_entries_matching(url=nil)
```

Comment:
maaaaybe, but I think it's OK as-is. If you think of it as "match at least this string/url fragment", then nil/empty string does match everything.

If I had near-term plans to upstream this I think this would be more convincing.